Cilium: Remove encryption state on NodeDelete events

jrfastab opened the pull request:
Delete encryption state when nodes are deleted otherwise the datapath will contain state for every node that existed.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/7776)
<!-- Reviewable:end -->

jrfastab committed:
cilium: remove encryption route and rules if crypto is disabled

Currently, we leave routing rules in place if a restart happens and
the newly started agent disables encryption. Its a bit more friendly
to remove the rules to cleanup state.

Signed-off-by: John Fastabend <john.fastabend@gmail.com>

jrfastab committed:
cilium: encryption, remove xfrm rules on nodeDelete events

Currently we are not deleting xfrm rules on nodeDelete events. Over
time this will continue to populate the kernel xfrm table so its
better if we remove stale entries.

Signed-off-by: John Fastabend <john.fastabend@gmail.com>

joestringer commented on pkg/datapath/linux/node.go:
```diff
@@ -732,6 +789,13 @@ func (n *linuxNodeHandler) NodeConfigurationChanged(newConfig datapath.LocalNode
 	n.updateOrRemoveNodeRoutes(prevConfig.AuxiliaryPrefixes, newConfig.AuxiliaryPrefixes)
 
+	if !newConfig.EnableIPSec {
+		err := n.removeEncryptRules()
```
Does this succeed if encryption was previously not enabled?

jrfastab committed:
cilium: route, delete ipsec rule must include mask

Deleting the IPsec rule was failing on IP6 addresses because the mask
value was not being included. Include it here to avoid leaving IP6
routes around after nodes are deleted.

Signed-off-by: John Fastabend <john.fastabend@gmail.com>

joestringer commented on pkg/datapath/linux/ipsec/ipsec_linux.go:
```diff
@@ -310,6 +342,10 @@ func UpsertIPsecEndpoint(local, remote *net.IPNet, dir IPSecDir) (uint8, error)
+	scopedLog := log.WithFields(logrus.Fields{
+		"remote-ip": remote.IP,
+	})
+	scopedLog.Info("Delete IPsec Endpoint")
```
How often will this occur? We're trying to get to a point where there is little to no `Info()` messages after startup. Maybe it's okay if this is likely to just be once per node add/remove event. (/cc @tgraf )

jrfastab replied:
Good catch lets just remove it. Its not so useful except when trying to find delete events but its better to debug the nodeDelete and k8s watcher at that point.

joestringer commented on pkg/mtu/detect_linux.go:
```diff
@@ -65,6 +66,9 @@ func autoDetect() (int, error) {
 	}
 
 	if mtu := link.Attrs().MTU; mtu != 0 {
+		if mtu < maxCiliumMTU {
```
isn't this the reverse of what the commit message is trying to say?

jrfastab replied:
typo thanks.
